blacktie/jatmibroker-xatmi/src/main/java/org/jboss/narayana/blacktie/jatmibroker/core/transport/hybrid/stomp/StompManagement.java

```diff
@@ -81,14 +81,17 @@ public static void send(Message message, OutputStream outputStream) throws IOExc
             toSend.append("\n");
         }
         toSend.append("\n");
-        outputStream.write(toSend.toString().getBytes());
+        synchronized (outputStream) {
+            outputStream.write(toSend.toString().getBytes());
```

Comment:
OK understood, this approach requires more changes than I anticipated:

> A socket will have a channel if, and only if, the channel itself was created via the SocketChannel.open or 
> ServerSocketChannel.accept methods.

So, yes, I agree with your original approach of synchronizing on the stream.